spec/system/publishers/publishers_can_archive_candidate_messages_spec.rb

```diff
@@ -0,0 +1,136 @@
+require "rails_helper"
+
+RSpec.describe "Publishers can archive and unarchive candidate messages", :js do
+  let(:organisation) { create(:school) }
+  let(:publisher) { create(:publisher) }
+  let(:jobseeker) { create(:jobseeker) }
+
+  let(:health_vacancy) { create(:vacancy, :live, publisher_organisation: organisation) }
+  let(:music_vacancy) { create(:vacancy, :live, publisher_organisation: organisation) }
+
+  let(:health_job_application) { create(:job_application, :status_submitted, vacancy: health_vacancy, jobseeker: jobseeker) }
+  let(:music_job_application) { create(:job_application, :status_submitted, vacancy: music_vacancy, jobseeker: jobseeker) }
+
+  let!(:health_conversation) { create(:conversation, job_application: health_job_application, archived: false) }
+  let!(:music_conversation) { create(:conversation, job_application: music_job_application, archived: false) }
+
+  before do
+    create(:jobseeker_message, conversation: health_conversation, sender: jobseeker)
+    create(:jobseeker_message, conversation: music_conversation, sender: jobseeker)
+    login_publisher(publisher: publisher, organisation: organisation)
+  end
+
```

Comment:
it's always nice to logout after each test - I found tests were possibly flakey untll I gave them all logout methods.